sdk/eventhub/Azure.Messaging.EventHubs/src/EventData.cs

```diff
@@ -272,6 +290,46 @@ public EventData(ReadOnlyMemory<byte> eventBody) : this(eventBody, lastPartition
+            DateTimeOffset? lastPartitionPropertiesRetrievalTime = null,
+            int? publishedSequenceNumber = null,
+            int? pendingPublishSequenceNumber = null,
+            long? pendingProducerGroupId = null,
+            short? pendingOwnerLevel = null) : this(new BinaryData(eventBody), properties, systemProperties, sequenceNumber, offset, enqueuedTime, partitionKey,
+                lastPartitionSequenceNumber, lastPartitionOffset, lastPartitionEnqueuedTime, lastPartitionPropertiesRetrievalTime, publishedSequenceNumber, pendingPublishSequenceNumber,
+                pendingProducerGroupId, pendingOwnerLevel)
+        {
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="EventData"/> class.
+        /// </summary>
+        ///
+        /// <param name="eventBody">The raw data as binary to use as the body of the event.</param>
+        /// <param name="properties">The set of free-form event properties to send with the event.</param>
+        /// <param name="systemProperties">The set of system properties received from the Event Hubs service.</param>
+        /// <param name="sequenceNumber">The sequence number assigned to the event when it was enqueued in the associated Event Hub partition.</param>
+        /// <param name="offset">The offset of the event when it was received from the associated Event Hub partition.</param>
+        /// <param name="enqueuedTime">The date and time, in UTC, of when the event was enqueued in the Event Hub partition.</param>
+        /// <param name="partitionKey">The partition hashing key applied to the batch that the associated <see cref="EventData"/>, was sent with.</param>
+        /// <param name="lastPartitionSequenceNumber">The sequence number that was last enqueued into the Event Hub partition.</param>
+        /// <param name="lastPartitionOffset">The offset that was last enqueued into the Event Hub partition.</param>
+        /// <param name="lastPartitionEnqueuedTime">The date and time, in UTC, of the event that was last enqueued into the Event Hub partition.</param>
+        /// <param name="lastPartitionPropertiesRetrievalTime">The date and time, in UTC, that the last event information for the Event Hub partition was retrieved from the service.</param>
+        /// <param name="publishedSequenceNumber">The publishing sequence number assigned to the event at the time it was successfully published.</param>
+        /// <param name="pendingPublishSequenceNumber">The publishing sequence number assigned to the event as part of a publishing operation.</param>
+        /// <param name="pendingProducerGroupId">The producer group identifier assigned to the event as part of a publishing operation.</param>
+        /// <param name="pendingOwnerLevel">The producer owner level assigned to the event as part of a publishing operation.</param>
+        ///
+        internal EventData(BinaryData eventBody,
```

Comment:
Do we need two of these, or can we get away with just one because it's internal?

Reply:
There are 28 uses across 8 files of the original (`ReadOnlyMemory<byte>`) constructor. For example, they'll provide eventBody like `new byte[] { 0x11, 0x22, 0x33 }`. To remove the original constructor and only use this one, I would have had to add `new BinaryData(...)` around all of those constructor calls. However, looking at it now, I realize this new constructor is only used in this class itself, which means I can make this on `private` and keep the original `internal`. I can't remove this new constructor because the previous constructor will always make a new instance of BinaryData, which doesn't work for scenarios like Clone or where someone would expect passing in BinaryData (via public constructor) would keep the same instance. So, the simplest (least widespread changes) was to keep the original constructor. If you want me to remove the original constructor, update all the calls to it, and keep this new one internal, I can do that. Otherwise, I'm going to make this new one private. Your call.

Reply:
I'm good with as-is.   It'd be nice to keep the surface smaller, but that's a lot of call sites to update, most of which I would assume to be tests.